sample-app/src/test/java/com/fabernovel/statefullayout/robot/robots.kt

```diff
@@ -0,0 +1,109 @@
+package com.fabernovel.statefullayout.robot
+
+import androidx.fragment.app.testing.FragmentScenario
+import androidx.fragment.app.testing.FragmentScenario.launchInContainer
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.ViewInteraction
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.Visibility
+import androidx.test.espresso.matcher.ViewMatchers.Visibility.GONE
+import androidx.test.espresso.matcher.ViewMatchers.Visibility.VISIBLE
+import androidx.test.espresso.matcher.ViewMatchers.hasChildCount
+import androidx.test.espresso.matcher.ViewMatchers.hasDescendant
+import androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import com.fabernovel.statefullayout.sample.R
+import com.fabernovel.statefullayout.sample.ui.empty.EmptyFragment
+import com.fabernovel.statefullayout.sample.ui.main.MainFragment
+import com.fabernovel.statefullayout.utils.withStatefulLayoutStateId
+
+open class TestRobot {
+    fun clickOn(resId: Int): ViewInteraction {
+        return onView(withId(resId))
+            .perform(ViewActions.click())
+    }
+
+    fun hasVisibility(resId: Int, visibility: Visibility): ViewInteraction {
+        return onView(withId(resId))
+            .check(matches(withEffectiveVisibility(visibility)))
+    }
+}
+
+open class StatefulLayoutRobot : TestRobot() {
```

Comment:
I think it's better to have each robot in its own file : Over time the robots will grow and more robots will be added

Reply:
Tests are only going to be about a single class with the same base actions. 
Each test have its own robot.

Reply:
Sure, but it doesn't mean that everything should be in the same file, especially if you want to implement more tests with more robots in the future : it will result in a file with 1000+ lines

And I think that you need to separate the two types of robots that you have here :
- you have a robot that implements the actions/assertions available for the statefulLayout
- you have robots that implement the interactions available on a fragment

-> the fragment robots should use the stateful robot (= composition, not inheritance)
-> And because they don't have the same purpose, they should be in separate files